src/modules/direction/store/directionSlice.ts

```diff
@@ -132,68 +161,98 @@ export const fetchExperts = (
   const { posts, meta } = getState().directions[direction.name].materials;
 
-  const response = await getPosts('latest-by-direction', {
-    params: {
-      direction: direction.id,
-      page: meta.pageNumber + 1,
-      size: LOAD_POSTS_LIMIT,
-    },
-  });
-
-  const fetchedPosts: IPost[] = response.data.content.map((post) => {
-    const author = _.pick(post.author, [
-      'avatar',
-      'firstName',
-      'id',
-      'lastName',
-      'mainInstitution',
-    ]);
-
-    const preview = _.truncate(post.content, {
-      length: 150, // TODO: use MAX_LEN constant
+  try {
+    const response = await getPosts('latest-by-direction', {
+      params: {
+        direction: direction.id,
+        page: meta.pageNumber + 1,
+        size: LOAD_POSTS_LIMIT,
+      },
+    });
+
+    const fetchedPosts: IPost[] = response.data.content.map((post) => {
+      const author = _.pick(post.author, [
+        'avatar',
+        'firstName',
+        'id',
+        'lastName',
+        'mainInstitution',
+      ]);
+
+      const preview = _.truncate(post.content, {
+        length: 150, // TODO: use MAX_LEN constant
+      });
+
+      return {
+        author,
+        mainDirection: DIRECTION_PROPERTIES[post.mainDirection.id.toString()],
+        postType: postTypeProperties[post.type.id.toString()],
+        title: post.title,
+        content: post.content,
+        preview,
+        createdAt: post.createdAt,
+      };
     });
 
-    return {
-      author,
-      mainDirection: DIRECTION_PROPERTIES[post.mainDirection.id.toString()],
-      postType: postTypeProperties[post.type.id.toString()],
-      title: post.title,
-      content: post.content,
-      preview,
-      createdAt: post.createdAt,
-    };
-  });
-
-  dispatch(
-    loadMaterials({
-      directionName: direction.name,
-      materials: {
-        posts: posts.concat(fetchedPosts),
-        meta: {
-          isLastPage: response.data.last,
-          isLoading: false,
-          pageNumber: response.data.number,
+    dispatch(
+      loadMaterials({
+        directionName: direction.name,
+        materials: {
+          posts: posts.concat(fetchedPosts),
+          meta: {
+            isLastPage: response.data.last,
+            loading: LoadingStatusEnum.succeeded,
+            pageNumber: response.data.number,
+            error: null,
+          },
         },
-      },
-    }),
-  );
+      }),
+    );
+  } catch (e) {
+    dispatch(
+      setMaterialsLoadingStatus({
+        direction,
+        status: LoadingStatusEnum.failed,
+        error: String(e),
+      }),
+    );
+  }
+};
+
+export const fetchInitialMaterials = (
+  direction: IDirection,
+): AppThunkType => (dispatch, getState) => {
+  const { posts } = getState().directions[direction.name].materials;
+
+  if (posts.length === 0) {
+    dispatch(
+      setMaterialsLoadingStatus({
+        direction,
+        status: LoadingStatusEnum.pending,
+      }),
+    );
+    dispatch(fetchMaterials(direction));
```

Comment:
why do we set loading only in case these's no posts but don't fetch the posts themselves?